src/NUnitFramework/framework/Constraints/Constraint.cs

```diff
@@ -54,8 +54,9 @@ protected Constraint(params object[] args)
 
             _displayName = new Lazy<string>(() =>
             {
-                var displayName = this.GetType().Name;
-                if (displayName.EndsWith("`1", StringComparison.Ordinal) || displayName.EndsWith("`2", StringComparison.Ordinal))
+                var type = this.GetType();
+                var displayName = type.Name;
+                if (type.IsGenericType)
```

Comment:
But was the portable build passing on your Mac? I so, then there may be more here than meets the eye.